byron/ledger/executable-spec/src/Ledger/Delegation.hs

```diff
@@ -736,3 +774,53 @@ maxCertsPerBlock groupedCerts
   = case groupedCerts of
       [] -> 0
       _  -> List.maximum (length <$> groupedCerts)
+
+
+--------------------------------------------------------------------------------
+-- Goblins instances
+--------------------------------------------------------------------------------
+
+deriveGoblin ''DCert
+
+
+--------------------------------------------------------------------------------
+-- AddShrinks instances
+--------------------------------------------------------------------------------
+
+deriveAddShrinks ''DCert
+
+
+--------------------------------------------------------------------------------
+-- SeedGoblin instances
+--------------------------------------------------------------------------------
+
+deriveSeedGoblin ''DSEnv
+deriveSeedGoblin ''DIState
+
+
+--------------------------------------------------------------------------------
+-- `GoblinData`s
+--------------------------------------------------------------------------------
+
+goblinData_DELEG_SDelegSFailure_SDelegFailure_EpochInThePast :: GoblinData Bool
+goblinData_DELEG_SDelegSFailure_SDelegFailure_EpochInThePast =
+  $(loadGoblinDataFromFilePath
+      "src/goblin_genomes/DELEG:_SDelegSFailure_(SDelegFailure_(EpochInThePast_(EpochDiff_{currentEpoch_=_Epoch_{unEpoch_=_1},_certEpoch_=_Epoch_{unEpoch_=_0}})))")
```

Comment:
These are funny file names!

Reply:
Yeah... they are the `show` of the PredicateFailure datatype, but substituting `'_'` for `' '`. The simplest hack I could come up with to make them valid filepaths 😅

Reply:
Why do they have specific values? E.g.: `Epoch_{unEpoch_=_0}`?

Reply:
I had to choose dummy values to train the goblins towards, but they are ignored by the `eqPF` predicate. It's a bit sloppy, though, and perhaps hashing the string would be tidier.